src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -76,6 +54,65 @@ public HttpApiHandler(DurableTaskExtension config, TraceWriter traceWriter)
             return response;
         }
 
+
+        internal async Task<HttpResponseMessage> CreateCheckStatusResponse(
+            HttpRequestMessage request,
+            string instanceId,
+            OrchestrationClientAttribute attribute,
+            int totalTimeout,
+            int retryTimeout)
+        {
+            GetClientResponseLinks(request, instanceId, attribute, out var statusQueryGetUri, out var sendEventPostUri,
+                out var terminatePostUri);
+
+            if (totalTimeout < retryTimeout)
+            {
+                throw new InvalidOperationException($"Total timeout {totalTimeout} should be bigger than {retryTimeout}");
```

Comment:
`ArgumentException` or `ArgumentOutOfRangeException` would be more appropriate in this context.